core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/plugins/ai-content-prompt.plugin.ts

```diff
@@ -63,14 +67,39 @@ export class AIContentPromptView {
         this.pluginKey = pluginKey;
         this.component = component;
 
-        this.component.instance.formSubmission.pipe(takeUntil(this.destroy$)).subscribe(() => {
-            this.editor.commands.closeAIPrompt();
-        });
-
-        this.component.instance.aiResponse.pipe(takeUntil(this.destroy$)).subscribe((content) => {
-            this.editor.commands.insertAINode(content);
-            this.editor.commands.openAIContentActions();
-        });
+        this.componentStore = this.component.injector.get(AiContentPromptStore);
+
+        /**
+         * Subscription to insert the AI Node and open the AI Content Actions.
+         */
+        this.componentStore.content$
+            .pipe(
+                takeUntil(this.destroy$),
+                filter((content) => !!content)
+            )
+            .subscribe((content) => {
+                this.editor
+                    .chain()
+                    .closeAIPrompt()
+                    .deleteSelection()
+                    .insertAINode(content)
+                    .openAIContentActions(DOT_AI_TEXT_CONTENT_KEY)
+                    .run();
+            });
+
+        /**
+         * Subscription to insert the text Content once accepted the generated content.
+         * Fired from the AI Content Actions plugin.
+         */
+        this.componentStore.vm$
```

Comment:
Listen the `acceptContent` only?

Reply:
is because I need `state.content` in the `subscribe`